### PR TITLE
fix(hooks): useProgress & usePlayback hooks

### PR DESCRIPTION
**Pull request by puckey**
- (breaking) fix: usePlaybackState was incorrectly returning `State.None` by default. Now it returns undefined while it is waiting on the async call to TrackPlayer to resolve. This also fixes the following issue where duration was incorrectly being reported as `0`: https://github.com/doublesymmetry/react-native-track-player/issues/1709
- optimization: avoid unnecessarily reevaluating the useEffect on every player state change by moving the `playerState === State.None` outside of the effect and including it as a dependency instead

### CODE REVIEW

**Comment by jspizziri** [comment]
@dcvz is this a significant enough breaking change to hold back the release?

@puckey is there a way of accomplishing this that doesn't introduce a breaking change?

File: src/hooks.ts [hunk 1]
```diff
@@ -131,7 +131,7 @@ export function useProgress(updateInterval = 1000) {
     return () => {
       mounted = false;
     };
-  }, [playerState, updateInterval]);
+  }, [isNone, updateInterval]);
 
   return state;
```
Comment (jspizziri) [inline src/hooks.ts:136]: @puckey would doing a:

```
return state || State.None;
```

Allow us to avoid the breaking change here?

**Review by jspizziri — CHANGES_REQUESTED** [review]
@puckey please see the comment I left. I'd like to find a way to do this without a breaking change. As we'd need to hold off until the next major version if not.

**Comment by gavrichards** [comment]
I'm keen to see this fixed asap, so I'll share that @puckey and I were discussing this in the discord yesterday, and the initial fix proposed which worked without any breaking changes is mentioned here:
https://discord.com/channels/567636850513018880/600714011318681610/1019166051461779467

File: src/hooks.ts (hunk 1 above)
Comment (puckey) [reply src/hooks.ts:136]: Not in this way, but there is slightly more convoluted way to fix this.. Will take a look at it later

**Comment by puckey** [comment]
How about something like this @jspizziri: https://github.com/doublesymmetry/react-native-track-player/pull/1723/commits/b895b9c9a183af7b25b8cc13c242501052316ed0

**Pull request merged**